lib/iris/config.py

```diff
@@ -154,3 +168,267 @@ def get_dir_option(section, option, default=None):
+
+        Example usages:
+
+        * Specify that we want to load a cube with dask parallel processing
+        using multiprocessing with six worker processes::
+
+            iris.options.parallel(scheduler='multiprocessing', num_workers=6)
+            iris.load('my_dataset.nc')
+
+        * Specify, with a context manager, that we want to load a cube with
+        dask parallel processing using four worker threads::
+
+            with iris.options.parallel(scheduler='threaded', num_workers=4):
+                iris.load('my_dataset.nc')
+
+        * Run dask parallel processing using a distributed scheduler that has
+        been set up at the IP address and port at ``192.168.0.219:8786``::
+
+            iris.options.parallel(scheduler='192.168.0.219:8786')
+
+        """
+        # Set `__dict__` keys first.
+        self.__dict__['_scheduler'] = scheduler
+        self.__dict__['scheduler'] = None
+        self.__dict__['num_workers'] = None
+        self.__dict__['dask_scheduler'] = None
+
+        # Set `__dict__` values for each kwarg.
+        setattr(self, 'scheduler', scheduler)
+        setattr(self, 'num_workers', num_workers)
+        setattr(self, 'dask_scheduler', self.get('scheduler'))
+
+        # Activate the specified dask options.
+        self._set_dask_options()
+
+    def __repr__(self):
+        msg = 'Dask parallel options: {}.'
+
+        # Automatically populate with all currently accepted kwargs.
+        options = ['{}={}'.format(k, v)
+                   for k, v in six.iteritems(self.__dict__)
+                   if not k.startswith('_')]
+        joined = ', '.join(options)
+        return msg.format(joined)
+
+    def __setattr__(self, name, value):
+        if value is None:
+            value = self._defaults_dict[name]['default']
+        attr_setter = getattr(self, 'set_{}'.format(name))
+        value = attr_setter(value)
+        super(Parallel, self).__setattr__(name, value)
+
+    @property
+    def _defaults_dict(self):
+        """
+        Define the default value and available options for each settable
+        `kwarg` of this `Option`.
+
+        Note: `'options'` can be set to `None` if it is not reasonable to
+        specify all possible options. For example, this may be reasonable if
+        the `'options'` were a range of numbers.
+
+        """
+        return {'_scheduler': {'default': None, 'options': None},
+                'scheduler': {'default': 'threaded',
+                              'options': ['threaded',
+                                          'multiprocessing',
+                                          'async',
+                                          'distributed']},
+                'num_workers': {'default': 1, 'options': None},
+                'dask_scheduler': {'default': None, 'options': None},
+                }
+
+    def set__scheduler(self, value):
+        return value
+
+    def set_scheduler(self, value):
```

Comment:
> they behave very similarly to the @property.setter in a more traditional class

But the _don't_ behave like a setter because they don't _set_ anything!